### PR TITLE
[IMIS] Use Jsoup instead of StringEscapeUtils to sanitize inputs and prevent XSS attacks #3389

### PR DESCRIPTION
**Pull request by Seal33**
More info can be found in #3389 and #3193 

Jsoup allows the use of custom whitelists to allow certain tags, while still preventing XSS attacks reliably. This is a cleaner and more standardized approach than that of PR #3220 , which will be reverted by this PR. Jsoup was already used in several spots throughout the code.

### CODE REVIEW

**Comment by MateStrysewske** [comment]
@Seal33 Please have a look at #3464; I had to create that issue because apparently `Jsoup.clean` can't deal with null values. I guess that we could and would run into a lot of problems with the current implementation in this PR; my suggestion would be to add some kind of a wrapper method to the API that calls Jsoup.clean if the given string is not null or just returns null again otherwise, and use that everywhere where you've added the call to Jsoup.clean.

**Comment by Seal33** [comment]
> @Seal33 Please have a look at #3464; I had to create that issue because apparently `Jsoup.clean` can't deal with null values. I guess that we could and would run into a lot of problems with the current implementation in this PR; my suggestion would be to add some kind of a wrapper method to the API that calls Jsoup.clean if the given string is not null or just returns null again otherwise, and use that everywhere where you've added the call to Jsoup.clean.

Oh, I didn't know that, I'm sorry we didn't catch that during our internal review. I think a wrapper method is a good idea. This would also allow us to create predefined Whitelists in the wrapper.

**Comment by syntakker** [comment]
@MateStrysewske I think it would make sense to do the sanitizing in the getters and setters of the Dto classes. This way sanitizing would be implemented in one central place and cannot be forgotten in the UI.

**Comment by MateStrysewske** [comment]
@syntakker That approach could make sense, although it would require developers to remember to add the sanitisation to every getter and setter they're adding. @MartinWahnschaffeSymeda do you have a preference here?

**Comment by MateStrysewske** [comment]
@syntakker @Seal33 Discussed this with Martin and we came to the conclusion that doing the sanitising in the getters and setters could make sense, but it's too big of a refactoring to be considered right now. We'll got with the solution that you've already implemented for now. I'll review this pull request after the next release.

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/HtmlHelper.java [hunk 1]
```diff
@@ -1,35 +1,53 @@
...
-		res = res.replaceAll("&lt;/ul&gt;", "</ul>");
-		// <li> Tags
-		res = res.replaceAll("&lt;li&gt;", "<li>");
-		res = res.replaceAll("&lt;/li&gt;", "</li>");
-		// <br> Tags
-		res = res.replaceAll("&lt;br&gt;", "<br>");
-		res = res.replaceAll("&lt;/br&gt;", "<br>");
-		res = res.replaceAll("&lt;br/&gt;", "<br>");
+	public static String cleanHtml(String string) {
+		if (string == null)
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/utils/HtmlHelper.java:30]: We're using brackets even for one-line statements; please refactor your code accordingly; also, you could use a shorter syntax for the return statements in this class (e.g. `return string == null ? "" : Jsoup.clean(string, Whitelist.none());`

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by MateStrysewske** [comment]
@Seal33 Can you solve the merge conflicts again? I'll approve the PR afterwards.

**Comment by Seal33** [comment]
@MateStrysewske The merge conflicts are now resolved. However, some tests fail during building. This is most likely caused by development failing tests right now, but I cannot guarantee that the merge itself did not introduce any new problems too. The branch did pass all tests before merging today.

**Comment by MateStrysewske** [comment]
@Seal33 Should be fine. We'll run manual tests on our system anyway. I don't think that anything that you've done in this PR should result in test failures.

Sorry, you'll have to do some conflict resolving again - I'll try to merge the PR asap afterwards so that you don't have to do it another time.

**Comment by Seal33** [comment]
@MateStrysewske Done. I also ran the tests locally with 0 failures.

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**